### PR TITLE
[4712] Reactify Site Tools - Item States

### PR DESCRIPTION
**Pull request by aachavarria**
craftercms/craftercms#4712

craftercms/craftercms#4721

Depends/Related craftercms/craftercms#4721

### CODE REVIEW

File: ui/app/src/components/ItemStatesManagement/ItemStatesManagement.tsx [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+
+  const onPathRegex$ = useDebouncedInput(
+    useCallback(
+      (keyword: string) => {
+        clearSelectedItems();
+        setDebouncePathRegex(keyword);
+      },
+      [setDebouncePathRegex]
+    ),
+    400
```
Comment (rart) [inline ui/app/src/components/ItemStatesManagement/ItemStatesManagement.tsx:130]: Wasn't this callback getting changed ignored by useDebouncedInput?

File: ui/app/src/components/ActionsBar/ActionsBar.tsx [hunk 2]
```diff
@@ -22,12 +22,16 @@ import { getPossibleTranslation } from '../../utils/i18n';
...
 
 const styles = makeStyles((theme) =>
   createStyles({
+    root: {
+      background: theme.palette.background.paper
+    },
     itemHeader: {
       display: 'flex',
-      background: '#e8f1ff'
+      background: palette.blue.highlight
```
Comment (rart) [inline ui/app/src/components/ActionsBar/ActionsBar.tsx]: `theme.palette.action.active`

File: ui/app/src/components/ItemStatesGrid/styles.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+export const useStyles = makeStyles((theme) =>
+  createStyles({
+    tableContainer: {},
+    ellipsis: {
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      whiteSpace: 'nowrap'
+    },
+    rowSelected: {
+      backgroundColor: palette.blue.highlight
```
Comment (rart) [inline ui/app/src/components/ItemStatesGrid/styles.ts]: theme.palette.action.active

File: ui/app/src/components/ItemStatesManagement/styles.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    },
+    helperText: {
+      marginLeft: 0
+    },
+    formControl: {
+      marginTop: '20px'
+    },
+    formLabel: {
+      color: theme.palette.text.primary,
+      textTransform: 'uppercase',
```
Comment (rart) [inline ui/app/src/components/ItemStatesManagement/styles.ts:67]: What's this for? I don't remember anything uppercased in the designs.

File: ui/app/src/components/PersistentDrawer/PersistentDrawer.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    root: (styles) => ({
+      flexShrink: 0,
+      ...styles.root
+    }),
+    drawerBody: (styles) => ({
+      width: '100%',
+      ...styles.drawerBody
+    }),
+    drawerPaper: (styles) => ({
+      top: 64,
```
Comment (rart) [inline ui/app/src/components/PersistentDrawer/PersistentDrawer.tsx]: See recent changes on ResizesableDrawer in regards to adding this top value.

File: ui/app/src/components/SetWorkflowStateDialog/SetWorkFlowStateDialogContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+import Switch from '@material-ui/core/Switch';
+import { useStyles } from './styles';
+import DialogFooter from '../Dialogs/DialogFooter';
+import SecondaryButton from '../SecondaryButton';
+import PrimaryButton from '../PrimaryButton';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import { CSSProperties } from '@material-ui/styles';
+
+export function SetWorkflowStateDialogContainer(props: SetWorkflowStateDialogProps) {
```
Comment (rart) [inline ui/app/src/components/SetWorkflowStateDialog/SetWorkFlowStateDialogContainer.tsx]: SetItemStateDialog*
Remember to change directory names too

File: ui/app/src/services/workflowStates.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+import { Observable } from 'rxjs';
+import { get, postJSON } from '../utils/ajax';
+import { map, mapTo, pluck } from 'rxjs/operators';
+import { toQueryString } from '../utils/object';
+import { SandboxItem } from '../models/Item';
+import { PagedArray } from '../models/PagedArray';
+import { createItemActionMap, createItemStateMap } from '../utils/content';
+import PaginationOptions from '../models/PaginationOptions';
+
+export function fetchItemStates(
```
Comment (rart) [inline ui/app/src/services/workflowStates.ts]: rename as `workflow.ts`

File: ui/app/src/components/PersistentDrawer/PersistentDrawer.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+      bottom: 0,
+      height: 'auto',
+      zIndex: theme.zIndex.appBar - 1,
+      ...styles.drawerPaper
+    })
+  })
+);
+
+export default function PersistentDrawer(props: ResizeableDrawerProps) {
```
Comment (rart) [inline ui/app/src/components/PersistentDrawer/PersistentDrawer.tsx]: Not sure I see much value in this component. 
Perhaps we could just use the ResizableDrawer with resize={false} or something of sorts.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/ItemStatesManagement/styles.ts (hunk 4 above)
Comment (aachavarria) [reply ui/app/src/components/ItemStatesManagement/styles.ts:67]: "Show Items In"

**Review by rart — APPROVED** [review]

**Pull request merged**